zilliqa/src/exec.rs

```diff
@@ -1368,11 +1424,18 @@ fn scilla_create(
         ));
     };
 
+    let init_data = ContractInit::new(init_data)?;
+
+    // We need to cache external libraries used in the current contract. Scilla checker needs to import them to check the contract.
+    cache_external_libraries(
+        &state.pre_state,
+        ext_libs_cache_dir,
+        &init_data.external_libraries()?,
+    )?;
     let check_output = match scilla.check_contract(&txn.code, gas, &init_data)? {
         Ok(o) => o,
         Err(e) => {
             warn!(?e, "transaction failed");
-            let gas = gas.min(e.gas_remaining);
```

Comment:
This has the effect of not charging for transactions which fail to check . I think we ought to charge for this ? @JamesHinshelwood ?

Reply:
The problem is if the check fails, the returned error, `e`, doesn't have gas_remaining:

```
CheckError { errors: [Error { start_location: Location { line: 2 }, error_message: "Ends in an error in state: 343.\n" }] }
```

Reply:
We need to charge a fixed value.

Reply:
Agree with @rrw-zilliqa - This code used to charge gas in the `let gas = gas.min(e.gas_remaining);` line and now we don't - Why? Clearly the Scilla server used to return `gas_remaining` if that code worked, so what changed?

Reply:
As I said, if the check fails the return value doesn't contain the gas_remaining. I'll revert my changes for now and will file another issue.